src/util.js

```diff
@@ -20,17 +20,30 @@ export function shouldRun() {
     })
   );
 
-  if (!excludeBranches.length) return true;
+  if (excludeBranches.length) {
+    const sourceBranch = github.context.payload.pull_request.head.ref;
 
-  const sourceBranch = github.context.payload.pull_request.head.ref;
-
-  const result = excludeBranches.some((p) => minimatch(sourceBranch, p));
-
-  if (result) {
-    core.notice("source branch matched the exclude pattern, exiting...");
+    if (excludeBranches.some((p) => minimatch(sourceBranch, p))) {
+      core.notice("source branch matched the exclude pattern, exiting...");
+      return false;
+    }
   }
 
-  return !result;
+  const excludeLabels = parseCSV(
+      core.getInput("exclude-labels", {
+        required: false,
+      })
+  );
+  if (excludeLabels.length) {
+    const labels = github.context.payload.pull_request.labels;
+    for (const label of labels) {
+      if (excludeLabels.includes(label.name)) {
+        core.notice("exclude label was found, exiting...");
+        return false;
+      }
+    }
```

Comment:
This for loop will throw `labels is not iterable` if they're not provided. Please coalesce to an empty array `|| []` when assigning `labels`.

Also just my two cents, but it would be more efficient and elegant to use `Array.some` instead of a for loop.

```suggestion
    const labels = github.context.payload.pull_request.labels || [];
    if (labels.some(({ name }) => excludeLabels.includes(name))) {
      core.notice("excluded label was found, exiting...");
      return false;
    }
```